app/assets/javascripts/pageflow/editor/models/uploaded_file.js

```diff
@@ -10,6 +10,13 @@ pageflow.UploadedFile = Backbone.Model.extend({
 
     this.configuration.i18nKey = this.i18nKey;
 
+    if (pageflow.editor.nextUploadTargetFile){
+      this.set({parent_file_id: pageflow.editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: pageflow.editor.nextUploadTargetFile.fileType().typeName});
+      this.collection = pageflow.editor.nextUploadTargetFile
+        .nestedFiles(this.fileType().collectionName);
+    }
+
```

Comment:
That's true, but it should be easy to pass a reference to the parent file when creating the `nestedFiles` collection. [`createForFiletype`](https://github.com/aviav/pageflow/blob/5ff59e547c3d24cd6c54d682155a4bfa6ab0f27e/app/assets/javascripts/pageflow/editor/collections/files_collection.js#L77) passes all options to the collection.